integration-tests/smoke/ccip_batching_test.go

```diff
@@ -0,0 +1,295 @@
+				state.Chains[destChain].OffRamp,
+				nil,
+				ccipocr3.NewSeqNumRange(startSeqNum[sourceChain1], endSeqNum[sourceChain1]),
+			)
+
+			mx.Lock()
+			errs = append(errs, err)
+			mx.Unlock()
+		}()
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			var err error
+			sourceChain2Report, err = ccdeploy.ConfirmCommitWithExpectedSeqNumRange(t,
+				e.Env.Chains[sourceChain2],
+				e.Env.Chains[destChain],
+				state.Chains[destChain].OffRamp,
+				nil,
+				ccipocr3.NewSeqNumRange(startSeqNum[sourceChain2], endSeqNum[sourceChain2]),
+			)
+
+			mx.Lock()
+			errs = append(errs, err)
+			mx.Unlock()
+		}()
+
+		t.Log("waiting for commit report")
+		wg.Wait()
+
+		for _, err := range errs {
+			require.NoError(t, err)
+		}
+
+		// the reports should be the same for both, since both roots should be batched within
+		// that one report.
+		require.Equal(t, sourceChain1Report, sourceChain2Report, "commit reports should be the same")
+
+		startSeqNum[sourceChain1] = endSeqNum[sourceChain1] + 1
+		endSeqNum[sourceChain1] = startSeqNum[sourceChain1] + ccipocr3.SeqNum(numMessages) - 1
+		startSeqNum[sourceChain2] = endSeqNum[sourceChain2] + 1
+		endSeqNum[sourceChain2] = startSeqNum[sourceChain2] + ccipocr3.SeqNum(numMessages) - 1
+	})
+}
+
+func sendMessages(
+	ctx context.Context,
+	t *testing.T,
+	sourceChain deployment.Chain,
+	sourceTransactOpts *bind.TransactOpts,
+	sourceOnRamp *onramp.OnRamp,
+	sourceRouter *router.Router,
+	sourceMulticall3 *multicall3.Multicall3,
+	destChainSelector uint64,
+	numMessages int,
+	receiver []byte,
+) error {
+	calls, totalValue := genMessages(
+		ctx,
+		t,
+		sourceRouter,
+		destChainSelector,
+		numMessages,
+		receiver,
+	)
+
+	// Send the tx with the messages through the multicall
+	tx, err := sourceMulticall3.Aggregate3Value(
+		&bind.TransactOpts{
+			From:   sourceTransactOpts.From,
+			Signer: sourceTransactOpts.Signer,
+			Value:  totalValue,
+		},
+		calls,
+	)
+	_, err = deployment.ConfirmIfNoError(sourceChain, tx, err)
+	if err != nil {
+		return err
+	}
+
+	// check that the message was emitted
+	iter, err := sourceOnRamp.FilterCCIPMessageSent(
+		nil, []uint64{destChainSelector}, nil,
+	)
+	if err != nil {
+		return err
+	}
+
+	// there should be numMessages messages emitted
+	for i := 0; i < numMessages; i++ {
+		if !iter.Next() {
+			return fmt.Errorf("expected %d messages, got %d", numMessages, i)
+		}
+		t.Logf("Message id of msg %d: %x", i, iter.Event.Message.Header.MessageId[:])
+	}
+
+	return nil
+}
+
+func genMessages(
+	ctx context.Context,
+	t *testing.T,
+	sourceRouter *router.Router,
+	destChainSelector uint64,
+	count int,
+	receiver []byte,
+) (calls []multicall3.Multicall3Call3Value, totalValue *big.Int) {
+	totalValue = big.NewInt(0)
+	for i := 0; i < count; i++ {
+		msg := router.ClientEVM2AnyMessage{
+			Receiver:     receiver,
+			Data:         []byte(fmt.Sprintf("hello world %d", i)),
+			TokenAmounts: nil,
+			FeeToken:     common.HexToAddress("0x0"),
+			ExtraArgs:    nil,
```

Comment:
I think we should add tokens to the test. It might not add a lot, but it would make the test more comprehensive and ensure a full message with all components works as expected. I noticed @0xsuryansh  used some helper functions for token sending in his PR that we could probably reuse here. We don’t need to verify the token transfers since Suryansh’s tests already cover that, but including tokens would still make the test more complete.

Reply:
Yeah, waiting for the tokens test to merge so I can use the helpers from there